airbyte-integrations/connectors/source-bing-ads/source_bing_ads/base_streams.py

```diff
@@ -193,6 +195,43 @@ def request_params(
             "ReturnAdditionalFields": self.additional_fields,
         }
 
+    @property
+    def state(self) -> Mapping[str, Any]:
+        return self._state
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        current_state_value = self._state.get(self.cursor_field, "")
+        if record_state_value := value[self.cursor_field]:
+            new_state_value = max(current_state_value, record_state_value)
+            self._state.update({self.cursor_field: new_state_value})
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+        **kwargs: Mapping[str, Any],
+    ) -> Iterable[Mapping[str, Any]]:
+        stream_state = stream_state or {}
+        next_page_token = None
+
+        while True:
+            params = self.request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+            response = self.send_request(params, customer_id=None)
+            for record in self.parse_response(response):
+                record = self.transform(record, stream_slice)
+                if sync_mode == SyncMode.incremental:
+                    if self.state.get(self.cursor_field, "") <= record.get(self.cursor_field):
```

Comment:
I think this logic does not work. Records are not sorted so we can assume the sequence of records:
* "LastModifiedTime": "2023-01-01T00:00:00.000000"
* "LastModifiedTime": "2020-01-01T00:00:00.000000"

Given an initial sync and this sequence of accounts, the second record would not be synced. Hence, I think we might have underestimated the complexity of implementing semi-incremental. I think we can exclude this for the certification and see if we implement it if a user request it.

Reply:
I agree, I will remove incremental from this pr